src/TextInputWidget.cc

```diff
@@ -170,6 +208,30 @@ FilteredTextEdit::textChanged()
         working_history_[history_index_] = toPlainText();
 }
 
+void
+FilteredTextEdit::receiveImage(const QPixmap &img, const QString &img_name)
+{
+        // Save image into temporary path to be loaded later.
+        QString img_path = QDir::tempPath() + '/' + img_name;
+        QSharedPointer<QFile> file{new QFile{img_path, this}};
+
+        if (!file->open(QIODevice::WriteOnly)) {
+                qDebug() << "Failed to open temporary image file:" << file->errorString();
+                previewDialog_->close();
+                return;
+        }
+        if (!img.save(file->fileName(), "PNG")) {
```

Comment:
Saving the file to disk remains an unnecessary pessimization, and it's further unclear how you are guaranteeing that the temporary files are deleted when no longer needed.

You can encode images without touching the disk using a [QImageWriter](https://doc.qt.io/qt-5/qimagewriter.html) (for example).